facia-tool/test/public/spec/validate.image.spec.js

```diff
@@ -1,122 +1,336 @@
+                .then(done.fail, err => {
+                    expect(err.message).toMatch(/does not have a valid crop/i);
+                    done();
+                });
+            });
+
+            it('fails if crop doesn\'t respect criteria', function (done) {
+                validate.gridInstance.getImage = () => {
+                    return Promise.resolve({
+                        data: {
+                            exports: [{
+                                id: 'image_crop',
+                                assets: [
+                                    { dimensions: { width: 5000, height: 100 } },
+                                    { dimensions: { width: 500, height: 10 } },
+                                    { dimensions: { width: 50, height: 1 } }
+                                ]
+                            }]
+                        }
+                    });
+                };
+
+                validate('http://grid.co.uk/1234567890123456789012345678901234567890?crop=image_crop', {
+                    minWidth: 100,
+                    maxWidth: 1000,
+                    widthAspectRatio: 5,
+                    heightAspectRatio: 4
+                })
+                .then(done.fail, err => {
+                    expect(err.message).toMatch(/does not have a valid crop/i);
+                    done();
+                });
+            });
+
+            it('gets the specified asset', function (done) {
+                validate.gridInstance.getImage = () => {
+                    return Promise.resolve({
+                        data: {
+                            exports: [{
+                                id: 'image_crop',
+                                assets: [
+                                    { dimensions: { width: 1400, height: 1400 } },
+                                    {
+                                        file: images.path('square.png'),
+                                        dimensions: { width: 140, height: 140 }
+                                    }
+                                ]
+                            }]
+                        }
+                    });
+                };
+
+                validate('http://grid.co.uk/1234567890123456789012345678901234567890?crop=image_crop', {
+                    minWidth: 100,
+                    maxWidth: 1000,
+                    widthAspectRatio: 1,
+                    heightAspectRatio: 1
+                })
+                .then(image => {
+                    expect(image.width).toBe(140);
+                    expect(image.height).toBe(140);
+                    expect(image.src).toMatch(/square\.png$/);
+                    expect(image.origin).toBe('http://media/image/1234567890123456789012345678901234567890');
+                })
+                .then(done)
+                .catch(done.fail);
+            });
+        });
+
+        describe('- link does not include crop id', function () {
```

Comment:
From the tests below it seems to pick the first acceptable export?

The reason I'm asking is because we're thinking of what to do when people drag a non-crop image into the different tools, so it might be nice if the behaviour was uniform.